packages/typegpu/tests/unmanagedIO.test.ts

```diff
@@ -1,245 +0,0 @@
-      .createBuffer(arrayOf(u32, 3), [1, 2, 3])
-      .$device(device);
-
-    expect(device.createBuffer).not.toHaveBeenCalled();
-    expect(tgpuBuffer.buffer).toBeDefined();
-
-    expect(device.createBuffer).toHaveBeenCalledWith({
-      size: 12,
-      usage: GPUBufferUsage.COPY_DST | GPUBufferUsage.COPY_SRC,
-      mappedAtCreation: true,
-    });
-    expect(tgpuBuffer.buffer.getMappedRange).toHaveBeenCalled();
-    expect(tgpuBuffer.buffer.unmap).toHaveBeenCalled();
-  });
-
-  it('should write to a mapped buffer', () => {
-    const device = mockDevice as unknown as GPUDevice;
-    mockBuffer.mapState = 'mapped';
-    const buffer = tgpu
-      .createBuffer(arrayOf(u32, 3), mockBuffer as unknown as GPUBuffer)
-      .$device(device);
-    tgpu.write(buffer, [1, 2, 3]);
-
-    expect(mockBuffer.getMappedRange).toHaveBeenCalled();
-    expect(mockBuffer.unmap).not.toHaveBeenCalled();
-  });
-
-  it('should write to a buffer', () => {
-    const device = mockDevice as unknown as GPUDevice;
-    const buffer = tgpu.createBuffer(arrayOf(u32, 3)).$device(device);
-
-    tgpu.write(buffer, [1, 2, 3]);
-
-    expect(device.queue.writeBuffer).toHaveBeenCalledWith(
-      buffer.buffer,
-      0,
-      new ArrayBuffer(0),
-      0,
-      12,
-    );
-  });
-
-  it('should map a mappable buffer before reading', async () => {
-    setupMocksForWrapped();
-    const device = mockDevice as unknown as GPUDevice;
-    mockBuffer.usage = GPUBufferUsage.MAP_READ | GPUBufferUsage.COPY_DST;
-    const buffer = tgpu
-      .createBuffer(arrayOf(u32, 3), mockBuffer as unknown as GPUBuffer)
-      .$device(device);
-    const data = await tgpu.read(buffer);
-
-    expect(mockStagingBuffer.mapAsync).not.toHaveBeenCalled();
-    expect(mockBuffer.mapAsync).toHaveBeenCalled();
-    expect(data).toBeDefined();
-  });
-
-  it('should read from a mapped buffer', async () => {
-    const device = mockDevice as unknown as GPUDevice;
-    mockBuffer.mapState = 'mapped';
-    const buffer = tgpu
-      .createBuffer(arrayOf(u32, 3), mockBuffer as unknown as GPUBuffer)
-      .$device(device);
-    const data = await tgpu.read(buffer);
-
-    expect(device.createBuffer).not.toHaveBeenCalled();
-    expect(data).toBeDefined();
-    expect(mockBuffer.getMappedRange).toHaveBeenCalled();
-    expect(mockBuffer.unmap).not.toHaveBeenCalled();
-  });
-
-  it('should read from a mappable buffer', async () => {
-    const device = mockDevice as unknown as GPUDevice;
-    mockBuffer.usage = mockBuffer.usage | GPUBufferUsage.MAP_READ;
-    const buffer = tgpu
-      .createBuffer(arrayOf(u32, 3), mockBuffer as unknown as GPUBuffer)
-      .$device(device);
-    const data = await tgpu.read(buffer);
-
-    expect(device.createBuffer).not.toHaveBeenCalled();
-    expect(data).toBeDefined();
-    expect(mockBuffer.getMappedRange).toHaveBeenCalled();
-    expect(mockBuffer.unmap).toHaveBeenCalled();
-  });
-
-  it('should read from a buffer', async () => {
```

Comment:
we don't have any tests for `read` (and `copyFrom`) no more :( 
maybe we could modify these instead of deleting them?